ertviz/controllers/multi_parameter_controller.py

```diff
@@ -40,37 +28,51 @@ def update_parameter_options(selected_ensembles):
             {"label": parameter_key, "value": parameter_key}
             for parameter_key in ensemble.parameters
         ]
-        return options
+        prev_selection = [
+            plot["name"] for plot in prev_plot_selection if plot["type"] == "parameter"
+        ]
+        return options, prev_selection
 
     @app.callback(
-        Output(parent.uuid("bincount-store"), "data"),
-        [Input(parent.uuid("hist-bincount"), "value")],
-        [State(parent.uuid("bincount-store"), "data")],
+        Output({"index": MATCH, "type": parent.uuid("bincount-store")}, "data"),
+        [Input({"index": MATCH, "type": parent.uuid("hist-bincount")}, "value")],
+        [State({"index": MATCH, "type": parent.uuid("bincount-store")}, "data")],
     )
-    def _update_bincount(hist_bincount, store_bincount):
+    def update_bincount(hist_bincount, store_bincount):
         if hist_bincount == store_bincount:
             raise PreventUpdate
         return hist_bincount
 
     @app.callback(
         [
             Output(
-                {"id": parent.uuid("parameter-scatter"), "type": parent.uuid("graph")},
+                {
+                    "index": MATCH,
+                    "id": parent.uuid("parameter-scatter"),
+                    "type": parent.uuid("graph"),
+                },
                 "figure",
             ),
-            Output(parent.uuid("hist-bincount"), "value"),
+            Output({"index": MATCH, "type": parent.uuid("hist-bincount")}, "value"),
         ],
         [
-            Input(parent.uuid("parameter-selector"), "value"),
-            Input(parent.uuid("hist-check"), "value"),
-            Input(parent.uuid("bincount-store"), "data"),
+            Input({"index": MATCH, "type": parent.uuid("hist-check")}, "value"),
+            Input(
+                {"index": MATCH, "type": parent.uuid("bincount-store")},
+                "modified_timestamp",
+            ),
+        ],
+        [
+            State(parent.uuid("ensemble-selection-store"), "data"),
+            State({"index": MATCH, "type": parent.uuid("parameter-id-store")}, "data"),
+            State({"index": MATCH, "type": parent.uuid("bincount-store")}, "data"),
         ],
-        [State(parent.uuid("ensemble-selection-store"), "data")],
     )
-    def _update_histogram(parameter, hist_check_values, bin_count, selected_ensembles):
+    def update_histogram(
+        hist_check_values, _, selected_ensembles, parameter, bin_count
+    ):
         if not selected_ensembles:
             raise PreventUpdate
-
         if dash.callback_context.triggered[0]["prop_id"].split(".")[0] == parent.uuid(
```

Comment:
The `parameter-selector` is not longer among inputs, so can go away I reckon.

Reply:
It has already been removed :shrug:

Reply:
I still see it there though:
```python
if dash.callback_context.triggered[0]["prop_id"].split(".")[0] == parent.uuid(
            "parameter-selector"
        ):
            bin_count = None
```

Reply:
removed now